### PR TITLE
Fix RotationPaletteEffect not working with player palettes

### LINKED ISSUE #13626: Water pixel rotation is broken on RA naval actors
Dunno since when exactly, but it's already been like this on last release.

### PR DESCRIPTION
**Pull request by reaperrr**
Closes #13626.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs [hunk 1]
```diff
@@ -106,11 +106,12 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 			}
 		}
 
-		static bool AnyPaletteNameStartsWith(HashSet<string> names, string prefix)
+		static bool AnyPaletteNameStartsWith(HashSet<string> names, string currentPalette)
 		{
 			// PERF: Avoid LINQ.
+			// Note: kvp.Key already includes a suffix if it's a player palette, so we need to check if the current palette starts with 'name', not the other way round
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs]: `name.StartsWith(prefix)` was a mistake/bug, so you don't need this comment.

File: OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs [hunk 2]
```diff
@@ -106,11 +106,12 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 			}
 		}
 
-		static bool AnyPaletteNameStartsWith(HashSet<string> names, string prefix)
+		static bool AnyPaletteNameStartsWith(HashSet<string> names, string currentPalette)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs]: I'd change this to something like `static bool startsWithAny(string name, HashSet<string> prefixes)`.

**Comment by reaperrr** [comment]
Updated.

**Review by abcdefg30 — APPROVED** [review]

File: OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs [hunk 3]
```diff
@@ -92,8 +92,8 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 
 			foreach (var kvp in palettes)
 			{
-				if ((info.Palettes.Count > 0 && !AnyPaletteNameStartsWith(info.Palettes, kvp.Key))
-					|| (info.ExcludePalettes.Count > 0 && AnyPaletteNameStartsWith(info.ExcludePalettes, kvp.Key)))
+				if ((info.Palettes.Count > 0 && !StartsWithAny(kvp.Key, info.Palettes))
+					|| (info.ExcludePalettes.Count > 0 && StartsWithAny(kvp.Key, info.Palettes)))
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs]: Typo: `info.ExcludePalettes`.  Right now you're checking against `info.Palettes` again.

**Review by pchote — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs (hunk 3 above)
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs]: Oops...

**Comment by reaperrr** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ba6243e86eea2bd06ce4b4625d460fc825268190

**Pull request merged**